### DISCUSSION TITLE
How can I send a notification to an email address without having a database record for the recipient?

### DISCUSSION THREAD

**Discussion by straversi** [excid3/noticed#418] [Q&A] [answered]
If I do:
```ruby
MyNotifier.with(record: thing).deliver('test@test.com')
```
I get an exception about how strings don't have base classes.

If I do:
```ruby
MyNotifier.with(record: thing).deliver(User.new(email: 'test@test.com'))
```
to try and get around that, I get a `null value in column "recipient_id" of relation "noticed_notifications" violates not-null constraint` because the User is not saved in the database.

Is there any way to do this, or am I out of luck here? Thank you!

**Comment by excid3** [comment, marked as answer]
In this case, you don't have a recipient object to deliver to, so you will want to use a bulk delivery.

I would use `MyNotifier.with(record: thing, email: "test@test.com").deliver` and use a Bulk Delivery method to send the email. You'll have to create a custom bulk email delivery method that sends the mailer as well.

**Reply by straversi** [reply]
Ahhh, so smart. Thank you!

**Reply by t2** [reply]
@straversi curious what your custom bulk delivery method looks like? I'm facing the same need.
